stage0/run.go

```diff
@@ -491,6 +492,19 @@ func Run(cfg RunConfig, dir string, dataDir string) {
 		args = append(args, "--local-config="+cfg.LocalConfig)
 	}
 
+	s1v, err := getStage1InterfaceVersion(dir)
+	if err != nil {
+		log.FatalE("error determining stage1 interface version", err)
+	}
+
+	if cfg.Hostname != "" {
+		if s1v > 1 {
```

Comment:
Dunno, it might be troublesome later. interfaceVersionThisThatAndOneMoreThingIfIRemeberedEverything.